main.js

```diff
@@ -5,17 +5,41 @@ const JSONStorage = require('node-localstorage').JSONStorage;
 const APPVERSION = require('./package.json').version;
 const https = require('https');
 const compareVersions = require('compare-versions');
+const minimist = require('minimist');
 
 const { app, BrowserWindow, ipcMain: ipc, Menu: menu, globalShortcut: gsc, shell } = electron;
 
+// slice args array differently
+// if calling electron directly in debug
+let arg_slice = 1
+
 if (process.env.NODE_ENV === 'development') {
   require('electron-debug')(); // eslint-disable-line global-require
+  args_slice = 2
+}
+
+let argv = minimist(process.argv.slice(arg_slice));
+
+// get data location
+const dataLocation = () =>{
+
+ let defaultLocation = process.env[(process.platform === 'win32') ? 'USERPROFILE' : 'HOME'] + '/.fromscratch' + (process.env.NODE_ENV === 'development' ? '/dev' : '');
+
+ if(argv.portable){
+  defaultLocation = process.cwd() +'/userdata'
+  if(argv.userdata){
+    if(typeof(argv.userdata) === 'string' ){
+      defaultLocation = argv.userdata
```

Comment:
nit, needs a `;` (in multiple places: line 29, 34, 38 too)

 (please check the eslint rules of the project, for this and spacing etc.)